### PR TITLE
fix(training): follow-up to multiple datasets

### PR DESCRIPTION
**Pull request by JPXKQX**
## Description

This PR contains minor updates.

- It supports the option to specify null for the `training_loss`, `validation_metrics` and `variable_groups` in the config when a dataset is only intended to be passed as input (= all dataset variables are forcing variables).
- It updates several type hints.

***As a contributor to the Anemoi framework, please ensure that your changes include unit tests, updates to any affected dependencies and documentation, and have been tested in a parallel setting  (i.e., with multiple GPUs). As a reviewer, you are also responsible for verifying these aspects and requesting changes if they are not adequately addressed. For guidelines about those please refer to https://anemoi.readthedocs.io/en/latest/***

By opening this pull request, I affirm that all authors agree to the [Contributor License Agreement.](https://github.com/ecmwf/codex/blob/main/Legal/contributor_license_agreement.md)

### CODE REVIEW

File: training/src/anemoi/training/train/tasks/diffusionforecaster.py [hunk 1]
```diff
@@ -303,28 +306,26 @@ def compute_dataset_loss_metrics(
...
             Whether in validation mode, and to calculate validation metrics, by default False
             If False, metrics will be empty
 
         Returns
         -------
-        tuple[torch.Tensor, dict, torch.Tensor]
+        tuple[dict[str, torch.Tensor], dict, list[dict[str, torch.Tensor]]]
             Loss value, metrics, and predictions (per step)
         """
-        loss = torch.zeros(1, dtype=next(iter(batch.values())).dtype, device=self.device, requires_grad=False)
```
Comment (anaprietonem) [inline training/src/anemoi/training/train/tasks/diffusionforecaster.py:326]: to check - this is this just cleaning some left out/outdated code?

Comment (JPXKQX) [reply training/src/anemoi/training/train/tasks/diffusionforecaster.py:326]: yes, correct, the loss created here is not used anywhere
